clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcrImageProvider.java

```diff
@@ -0,0 +1,188 @@
+      throw new Error("More than 1 image has this " + (isTag ? "tag" : "digest") + "!  We can't handle this in the POC!");
+    } else if (imagesWithThisIdentifier.size() == 0) {
+      throw new Error(String.format("No image with the " + (isTag ? "tag" : "digest") + " %s was found.", identifier));
+    }
+
+    ImageDetail matchedImage = imagesWithThisIdentifier.get(0);
+
+    EcsDockerImage ecsDockerImage = new EcsDockerImage();
+    ecsDockerImage.setRegion(region);
+    ecsDockerImage.addAmiForRegion(region, matchedImage.getImageDigest());
+    ecsDockerImage.setAttribute("creationDate", matchedImage.getImagePushedAt());
+    ecsDockerImage.setImageName(buildFullDockerImageUrl(matchedImage.getImageDigest(),
+      matchedImage.getRegistryId(),
+      matchedImage.getRepositoryName(),
+      region));
+
+    return Collections.singletonList(ecsDockerImage);
+  }
+
+  private boolean imageFilter(ImageDetail imageDetail, String identifier, boolean isTag) {
+    return isTag ?
+      imageDetail.getImageTags() != null && imageDetail.getImageTags().contains(identifier) :
+      imageDetail.getImageDigest().equals(identifier);
+  }
+
+  private NetflixAmazonCredentials getCredentials(String accountId) {
+    for (AccountCredentials credentials : accountCredentialsProvider.getAll()) {
+      if (credentials instanceof NetflixAmazonCredentials) {
+        NetflixAmazonCredentials amazonCredentials = (NetflixAmazonCredentials) credentials;
+        if (amazonCredentials.getAccountId().equals(accountId)) {
+          return amazonCredentials;
+        }
+      }
+    }
+    throw new NotFoundException(String.format("AWS account %s was not found.  Please specify a valid account name", accountId));
+  }
+
+
+  private boolean isValidRegion(NetflixAmazonCredentials credentials, String region) {
+    return credentials.getRegions().stream()
+      .map(AmazonCredentials.AWSRegion::getName)
+      .anyMatch(region::equals);
+  }
+
+  private boolean isValidEcrUrl(String imageUrl) {
+    imageUrl = imageUrl.replace("http://", "").replace("https://", "");
+    Matcher matcher = REPOSITORY_URI_PATTERN.matcher(imageUrl);
+    return matcher.find();
+  }
+
+  private String extractAwsAccountId(String imageUrl) {
+    return extractString(ACCOUNT_ID_PATTERN, imageUrl, 1,
+      "The repository URI provided does not contain a proper account ID.");
+  }
+
+  private String extractEcrRepositoryName(String imageUrl) {
+    return extractString(REPOSITORY_NAME_PATTERN, imageUrl, 1,
+      "The repository URI provided does not contain a proper repository name.");
+  }
+
+  private String extractAwsRegion(String imageUrl) {
+    return extractString(REGION_PATTERN, imageUrl, 0,
+      "The repository URI provided does not contain a proper region.");
+  }
+
+  private String extractString(Pattern pattern, String imageUrl, int group, String error) {
+    Matcher matcher = pattern.matcher(imageUrl);
+    if (!matcher.find()) {
+      throw new Error(error);
```

Comment:
What is `Error`? Do you think we should have different exceptions based on the condition? I suspect if we need to do error handling, we wouldn't want to have a `Error`, as that'd be the same anti-pattern of doing a catch on `Exception` or `Throwable` -- too broad.

Reply:
Changed it to `IllegalArgumentException` - it comes down to the passed in (ecr) `url` being improper.